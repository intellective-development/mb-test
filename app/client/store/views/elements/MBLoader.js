// @flow
import * as React from 'react';
import bindClassNames from '../../../shared/utils/bind_classnames';
import styles from './MBLoader.scss';

const cn = bindClassNames(styles);

type Props = {
  className?: string
}

const MBLoader = ({ className }: Props) => {
  return (
    <div className={className}>
      <div className={cn('MBLoader_Ball', 'MBLoader_Ball__1')} />
      <div className={cn('MBLoader_Ball', 'MBLoader_Ball__2')} />
      <div className={cn('MBLoader_Ball', 'MBLoader_Ball__3')} />
    </div>
  );
};

export default MBLoader;
