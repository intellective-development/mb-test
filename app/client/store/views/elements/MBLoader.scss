@import "../style/variables";

$base-diameter: 6px;
$animation-duration: 1000ms;
$step-duration: $animation-duration / 4;
$ball-spacing: 3px;

// There are four steps in this animation:
// o o o
// O o o
// o O o
// o o O
// We accomplish this by staggering the starting times of each element and adding equivalent dead space to the end of animation,
// such that while the second circle is active (25%) the first is back to normal (50%) and the third has yet to begin (0%).

@keyframes ball-grow {
  0% { transform: scale(1); }
  25% { transform: scale(1.5); }
  50% { transform: scale(1); }
  100% { transform: scale(1); }
}

.MBLoader_Ball {
  display: inline-block;
  background-color: $brand-light-grey;
  border-radius: $base-diameter;
  width: $base-diameter;
  height: $base-diameter;
  animation: ball-grow $animation-duration linear infinite;
}

.MBLoader_Ball__1 {
  animation-delay: 0ms;
  margin-right: $ball-spacing;
}

.MBLoader_Ball__2 {
  animation-delay: $step-duration; // we offset this by 1 step
  margin-left: $ball-spacing;
  margin-right: $ball-spacing;
}

.MBLoader_Ball__3 {
  animation-delay: $step-duration * 2; // we offset this by 2 steps
  margin-left: $ball-spacing;
}
