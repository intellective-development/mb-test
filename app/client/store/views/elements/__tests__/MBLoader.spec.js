
import React from 'react';

import MBLoader from '../MBLoader';

describe('MBLoader', () => {
  it('renders', () => {
    expect(render(
      <MBLoader />
    )).toMatchSnapshot();
  });
});
