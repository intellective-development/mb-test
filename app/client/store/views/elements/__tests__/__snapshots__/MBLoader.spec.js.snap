// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MBLoader renders 1`] = `
<div>
  <div
    class="MBLoader_Ball MBLoader_Ball__1"
  />
  <div
    class="MBLoader_Ball MBLoader_Ball__2"
  />
  <div
    class="MBLoader_Ball MBLoader_Ball__3"
  />
</div>
`;
